WordPress/Sniffs/WP/PreparedSQLSniff.php

```diff
@@ -76,7 +76,8 @@ public function process( PHP_CodeSniffer_File $phpcsFile, $stackPtr ) {
 
 				$string = str_replace( '$wpdb', '', $tokens[ $i ]['content'] );
 
-				if ( false !== strpos( $string, '$' ) ) {
+				// Check if the $ is followed by a valid name, and that it is not preceded by an escape sequence.
+				if ( preg_match( '/(\\\\*)\$\w+/', $string, $matches ) && strlen( $matches[1] ) % 2 === 0 ) {
```

Comment:
Can you add some tests that demonstrate that this correctly handles the case where the variable is proceeded by an even number of backslashes and therefore is not escaped?

Also, what happens if there are multiple matches in a single string? I think if the first one is escaped the second one will be ignored, will it not? Tests please.